src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -1299,6 +1345,14 @@ void BlockchainLMDB::open(const std::string& filename, const int db_flags)
   LOG_PRINT_L2("Setting m_height to: " << db_stats.ms_entries);
   uint64_t m_height = db_stats.ms_entries;
 
+  MDB_stat rct_distr_stats;
+  if ((result = mdb_stat(txn, m_rct_distribution, &rct_distr_stats)))
+    throw0(DB_ERROR(lmdb_error("Failed to query m_rct_distribution: ", result).c_str()));
+  if (rct_distr_stats.ms_entries > m_height)
+    throw0(DB_ERROR("Invalid rct_distribution height"));
+  m_rct_distribution_offset = m_height - rct_distr_stats.ms_entries;
+  MDEBUG("rct distribution offset: " << m_rct_distribution_offset);
```

Comment:
I think `m_rct_distribution_offset` is given value only when A) the DB gets converted from v2 to v3, or B) the DB is opened. I.e. it doesn't get initialized when starting a full sync from scratch.